### DISCUSSION TITLE
Include directive inside ipsec.secrets is not working.

### DISCUSSION THREAD

**Discussion by acrolink** [strongswan/strongswan#570] [Q&A] [answered]
I am facing a strange problem. I try to include a secrets file inside `ipsec.secrets` like this:

`include /etc/ipsec.secrets.files/10_12_10_0.secrets`

but it doesn't work, the client fails to authenticate.

**Comment by tobiasbrunner** [comment]
Read the log and check if the secrets are actually loaded.

**Comment by acrolink** [comment, marked as answer]
I had to whitelist the custom path inside:

`/etc/apparmor.d/usr.lib.ipsec.charon`

Now it is working.
